src/engine/QueryPlanner.cpp

```diff
@@ -1610,7 +1677,8 @@ vector<vector<QueryPlanner::SubtreePlan>> QueryPlanner::fillDpTab(
         continue;
       }
       dpTab[k - 1].insert(dpTab[k - 1].end(), newPlans.begin(), newPlans.end());
-      applyFiltersIfPossible(dpTab.back(), filters, k == tg._nodeMap.size());
+      applyFiltersIfPossible(dpTab.back(), filters,
+                             k == (tg._nodeMap.size() + children.size()));
```

Comment:
swap arguments to `==` to match the order in the other `applyFiltersIfPossible()` call

Reply:
done